oak_functions/loader/src/server.rs

```diff
@@ -249,6 +280,22 @@ impl WasmState {
         .context("failed to instantiate Wasm module")?
         .assert_no_start();
 
+        let alloc = instance
+            .export_by_name(ALLOC_FUNCTION_NAME)
+            .context("could not find Wasm `alloc` export")?
+            .as_func()
+            .cloned()
+            .context("could not interpret Wasm `alloc` export as function")?;
+        let expected_signature = wasmi::Signature::new(&[ValueType::I32][..], Some(ValueType::I32));
+        if alloc.signature() != &expected_signature {
```

Comment:
Do we have a similar check for the `main` function too? If not, do we need that?

Reply:
Added in https://github.com/project-oak/oak/pull/2010